tests/kafkatest/tests/streams/streams_multiple_rolling_upgrade_test.py

```diff
@@ -0,0 +1,114 @@
+                                                                    'sum': {'partitions': 5, 'replication-factor': 1},
+                                                                    'dif': {'partitions': 5, 'replication-factor': 1},
+                                                                    'cnt': {'partitions': 5, 'replication-factor': 1},
+                                                                    'avg': {'partitions': 5, 'replication-factor': 1},
+                                                                    'wcnt': {'partitions': 5, 'replication-factor': 1},
+                                                                    'tagg': {'partitions': 5, 'replication-factor': 1}
+                                                                })
+
+        self.driver = StreamsSmokeTestDriverService(test_context, self.kafka)
+        self.processor_1 = StreamsSmokeTestJobRunnerService(test_context, self.kafka)
+        self.processor_2 = StreamsSmokeTestJobRunnerService(test_context, self.kafka)
+        self.processor_3 = StreamsSmokeTestJobRunnerService(test_context, self.kafka)
+
+        # always add new version as the SECOND TO LAST version before DEV_BRANCH
+        self.streams_upgrade_versions = [str(LATEST_0_10_2), str(LATEST_0_11_0), str(LATEST_1_0), str(DEV_BRANCH)]
+
+        # already on trunk version at end of upgrades so get rid of it
+        self.streams_downgrade_versions = self.streams_upgrade_versions[:-1]
+        self.streams_downgrade_versions.reverse()
+
+        self.processors = [self.processor_1, self.processor_2, self.processor_3]
+
+        self.started = False
+
+    def setUp(self):
+        self.zk.start()
+
+    def upgrade_and_verify_start(self, processors, to_version):
+        for processor in processors:
+            self.logger.info("Updating node %s to version %s" % (processor.node.account, to_version))
+            node = processor.node
+            if self.started:
+                processor.stop()
+                self.wait_for_verification(processor, "SMOKE-TEST-CLIENT-CLOSED", processor.STDOUT_FILE)
+            node.version = KafkaVersion(to_version)
+            processor.start()
+            self.wait_for_verification(processor, "initializing processor: topic", processor.STDOUT_FILE)
+
+        self.started = True
+
+    def update_processors_and_verify(self, versions):
+        for version in versions:
+            self.upgrade_and_verify_start(self.processors, version)
+        self.run_data_and_verify()
+
+    def run_data_and_verify(self):
+        self.driver.start()
+        self.wait_for_verification(self.driver, "ALL-RECORDS-DELIVERED", self.driver.STDOUT_FILE)
+        self.driver.stop()
+
+    @cluster(num_nodes=9)
+    @parametrize(broker_version=str(LATEST_0_10_2))
+    @parametrize(broker_version=str(LATEST_0_11_0))
+    @parametrize(broker_version=str(LATEST_1_0))
+    @parametrize(broker_version=str(DEV_BRANCH))
```

Comment:
Could we do `@matrix(broker_version =["...", ...}` as we did in simple benchmark?

Reply:
ack